main.sh

```diff
@@ -50,21 +83,33 @@ if [ -d $TEMPLATE_PATH ]; then
       exit 1
     else
 
-      git clone $TEMPLATE_PATH $DESTINATION &>/dev/null
+	  # Retain ref to template project so clones can pull in changes later. 
+	  # Perhaps rsync? that would look at dates, but maybe store the map in the config file
+      git clone $TEMPLATE_PATH $DESTINATION &>/dev/null 
       CLONE_RESULT=$?
       if [ $CLONE_RESULT -ne 0 ]; then
         echo "The template could not be cloned."
         exit 1
 
       else
-
+	    # this tracks the ref of the template (as mentioned in future options)
+        config_source=~/.start-project/config #  ${XDG_CONFIG_HOME}/start-project/config
+	    if [[ $(grep -e ${DESTINATION} $config_source ) ]]; then
+	      sed -ie "s%${DESTINATION}=%${DESTINATION}=$\(git --work-tree $DESTINATION rev-parse --short HEAD\)%" $config_source
+	    else
+	      echo "${DESTINATION}=$(git --work-tree $DESTINATION rev-parse --short HEAD)" >> $config_source
+        fi
+		# Do we need to cd? often this is where things could break, if a script
+		# forgets to cd back, it also allows for a path to be specified
         cd $DESTINATION
+		post_setup
         # Replace the placeholders with the project name.
-        # TODO: Note dependency on Ag.
-        ag PROJECT_NAME -l | xargs sed -e "s/PROJECT_NAME/$DESTINATION/g" -i ''
-
+        # TODO: Note dependency on Ag. could also do `eval ${search_cmd} | xargs....`
+        ag PROJECT_NAME -l ${DESTINATION} | xargs -I{} -P 4 sed -ie "s/PROJECT_NAME/$DESTINATION/g" {}
+		#just for fun, awk, related to future options ag alternatives (grep works too!)
```

Comment:
yeah `ag` typically isn't on many machines, that doesn't mean I don't think `ag` is cool or anything, it is, and its got a lot of features. I think if there was a need for `ag` or `ripgrep` then use that, otherwise, just stick with `grep` till the need arises?

Reply:
Yeah, probably a good idea.  Ag is definitely overkill for this!